crates/sqllib/src/array.rs

```diff
@@ -16,6 +17,25 @@ pub fn to_arrayN<T>(data: Option<Vec<T>>) -> Option<Array<T>> {
     data.map(|data| to_array(data))
 }
 
+#[doc(hidden)]
+pub fn sort<F, G, T, D>(mut data: Vec<T>, _cmp: F, proj: G) -> Array<D>
+where
+    F: CmpFunc<T>,
+    G: FnMut(T) -> D,
+{
+    data.sort_by(|a, b| F::cmp(a, b));
+    Arc::new(data.into_iter().map(proj).collect::<Vec<D>>())
```

Comment:
this will reallocate the entire array, you may want to use [unstable sort inplace](https://doc.rust-lang.org/std/primitive.slice.html#method.sort_unstable)?